virtManager/delete.py

```diff
@@ -543,5 +548,21 @@ def _vm_active_status(self):
     def _delete_vm(self, vm, undefine):
         pass
 
+    def _delete_disks(self, vm, paths, conn, meter):
+        storage_errors = []
+        if not paths:
+            vm.remove_device(self.disk)
+            return storage_errors
+        for path in paths:
+            try:
+                log.debug("Deleting path: %s", path)
+                meter.start(text=_("Deleting path '%s'") % path)
+                self._async_delete_dev(vm, conn, path, meter)
+            except Exception as e:
+                storage_errors.append((str(e),
+                                          "".join(traceback.format_exc())))
+            meter.end(0)
+        return storage_errors
```

Comment:
This block is now duplicated twice. It should be shared. Also vm.remove_device is only called when no paths are selected for deletion? In this case it looks like we should call it unconditionally, right?